Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/States/StateManager.cs

```diff
@@ -0,0 +1,422 @@
+            }
+
+            return state;
+        }
+
+        /// <summary>
+        /// Gets and sets a Core Interaction State to On and invokes the OnStateActivated event. Setting a 
+        /// state on changes state value to 1.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction state to set to on</param>
+        /// <returns>The Core Interaction state that was set to on</returns>
+        public InteractionState SetStateOn(CoreInteractionState coreState)
+        {
+            return SetStateOn(coreState.ToString());
+        }
+
+        /// <summary>
+        /// Gets and sets a state to Off and invokes the OnStateDeactivated event.  Setting a 
+        /// state off changes the state value to 0.
+        /// </summary>
+        /// <param name="stateName">The name of the state to set to off</param>
+        /// <returns>The state that was set to off</returns>
+        public InteractionState SetStateOff(string stateName)
+        {
+            InteractionState state = GetState(stateName);
+
+            if (state != null)
+            {
+                if (state.Value != 0)
+                {
+                    state.Value = 0;
+                    state.Active = 0;
+
+                    // If the only state in active states is going to be removed, then activate the default state
+                    if (activeStates.Count == 1 && activeStates.First() == state)
+                    {
+                        SetStateOn(CoreInteractionState.Default);
+                    }
+
+                    // We need to save the last state active state so we can add transitions 
+                    OnStateDeactivated.Invoke(state, activeStates.Last());
+
+                    activeStates.Remove(state);
+                }  
+            }
+            else
+            {
+                Debug.LogError($"The {stateName} state is not being tracked, add this state using AddState(state) to set it");
+            }
+
+            return state;
+        }
+
+        /// <summary>
+        /// Gets and sets a Core Interaction State to Off and invokes the OnStateDeactivated event.  Setting a 
+        /// state off changes the state value to 0.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction state to set to off</param>
+        /// <returns>The Core Interaction state that was set to off</returns>
+        public InteractionState SetStateOff(CoreInteractionState coreState)
+        {
+            return SetStateOff(coreState.ToString());
+        }
+
+        /// <summary>
+        /// Adds a new Core Interaction State to track.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction State to add</param>
+        /// <returns>The newly added Core Interaction State</returns>
+        public InteractionState AddCoreState(CoreInteractionState coreState)
```

Comment:
There's quite a bit of code overlap between this method and AddCoreState(InteractionState), can you re-use the code by for example doing this?

```
public InteractionState AddCoreState(CoreInteractionState coreState)
{
    AddCoreState(coreState.Name);
}
```

Reply:
Removed AddCoreState.  The state manager now only contains `AddNewState(string stateName)` instead of both `AddNewState` and `AddCoreState`.